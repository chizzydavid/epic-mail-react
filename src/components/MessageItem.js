import React from 'react'
import PropTypes from 'prop-types';

const MessageItem = (props) => {
  const imgUrl = `https://res.cloudinary.com/chizzydavid/image/upload/`;
  
  const {
    message_id, subject, status, email, sender_id, message, photo 
  } = props

	const id = Number(message_id);
  const msgBody = message.length >= 70 ? `${message.substr(0, 70)}...` : message;
  const type = Number(2) === sender_id ? 'sent' : 'received';
  const msgStatus = status.replace(status[0], status[0].toUpperCase());
  const userImg = `${imgUrl}${photo}`;
  return (
		<div className="message" data-message-id={id} data-message-type={type} data-message-status={status}>

			<div className="message-img">
				<img src={userImg} id="msg-sender-img"/>
			</div>

			<div className="message-text">
				<h4 className={status == 'unread' ? 'message-title unread' : 'message-title'}> {subject}	</h4>
	
				<div className="msg-body"> 
					<p className="msg-excerpt">{msgBody}</p>
					<div className="msg-details">
						<p>Status: <span className={'status-' + status + 'msg-status'}>{msgStatus} </span> </p>

						<div className="msg-buttons">
							{status === 'draft' ? (
								<React.Fragment>
									<i id={id} data-draft-receiver={email} className="edit-draft fa fa-edit"></i>
									<p className="hide">  {message}  </p>									
								</React.Fragment>
							) : '' }
							
							{type === 'sent' && status !== 'draft'
							? (<i id={id} className="retract-message fa fa-undo"></i>) : '' }
							<i id="${id}" data-message-type="${type}" className="delete-message fa fa-trash"></i>
						</div>            
					</div> 
				</div>    
			</div>
		</div>
  )
};

MessageItem.propTypes = {
  message_id: PropTypes.number.isRequired,
  subject: PropTypes.string.isRequired,
  email: PropTypes.string,
  sender_id: PropTypes.number.isRequired,
  message: PropTypes.string.isRequired,
  status: PropTypes.string,
  photo: PropTypes.string.isRequired
}

export default MessageItem
