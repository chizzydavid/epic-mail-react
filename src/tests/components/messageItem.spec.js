import React from 'react';
import { shallow } from 'enzyme';
import MessageItem from '../../components/MessageItem';

function shallowSetup() {
  const props = {
    message_id: 1,
    subject: 'How to make test',
    sender_id: 2,
    status: 'sent',
    message: 'fastest way to make tests pass',
    photo: "https://www.photos.com/a_photo",
  }
  const enzymeWrapper = shallow(<MessageItem {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Shallow rendered Message Item', () => {
  it('should render message item component', () => {
    const { enzymeWrapper } = shallowSetup();
    expect(enzymeWrapper).toMatchSnapshot();
  });
});

