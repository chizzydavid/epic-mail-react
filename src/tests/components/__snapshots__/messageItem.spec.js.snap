// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered Message Item should render message item component 1`] = `ShallowWrapper {}`;
